### PR TITLE
[Fix] fix the bug val loss log not show using IterBasedRunner

### PR DESCRIPTION
**Pull request by teamwong111**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Related Issue: https://github.com/open-mmlab/mmcv/issues/1163
Fix the bug val loss log not show using `IterBasedRunner`

## Modification

Impl `after_val_iter` function in `LoggerHook`. It should be noted that this function only works when the corresponding value of `workflow.val` is greater than the log config `interval` or `len(data_loader)`

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

**Before PR**:

- [x] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) to create this PR.
- [x] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) are used to fix the potential lint issues.
- [x] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [x] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [x] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [x] If the modification has potential influence on downstream or other related projects, this PR should be tested with some of those projects, like MMDet or MMCls.
- [x] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

**Comment by CptCaptain** [comment]
Does this PR have any chance of getting merged anytime soon?
Having to choose between using an iteration based schedule and logging validation losses seems... suboptimal, especially when the rest of the library is so great to use.
